apps/job-server/src/jellyfin/sync/items.ts

```diff
@@ -180,31 +181,187 @@ async function syncLibraryItems(
             metrics.incrementErrors();
           }
         },
-        { concurrency: options.itemConcurrency }
+        { concurrency: options.itemConcurrency },
       );
 
       startIndex += jellyfinItems.length;
       hasMoreItems = startIndex < totalCount && jellyfinItems.length > 0;
 
       console.log(
-        `Processed batch for library ${libraryId}: ${startIndex}/${totalCount} items`
+        `Processed batch for library ${libraryId}: ${startIndex}/${totalCount} items`,
       );
     } catch (error) {
       console.error(
         `Error fetching items page for library ${libraryId}:`,
-        error
+        error,
       );
       metrics.incrementErrors();
       break; // Stop processing this library on API error
     }
   }
 }
 
+/**
+ * Check if two items have matching provider IDs
+ */
+function hasMatchingProviderIds(
+  providerIds1: Record<string, string> | null,
+  providerIds2: Record<string, string> | null,
+): boolean {
+  if (!providerIds1 || !providerIds2) {
+    return false;
+  }
+
+  const allProviders = new Set(
+    [...Object.keys(providerIds1), ...Object.keys(providerIds2)].filter(
+      Boolean,
+    ),
+  );
+
+  for (const provider of allProviders) {
+    if (providerIds1[provider] === providerIds2[provider]) {
+      return true;
+    }
+  }
```

Comment:
If both records have undefined for the same provider key, this returns true incorrectly. Should validate both values are truthy:
```
for (const provider of allProviders) {
  const val1 = providerIds1[provider];
  const val2 = providerIds2[provider];
  if (val1 && val2 && val1 === val2) {
    return true;
  }
}
```